### PR TITLE
libgettext: fix hooks

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **libgettext/***

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
help me @madebr, you're my only hope !
I could reproduce the failure localy, and config.log has:
```
configure:4950: checking for x86_64-w64-mingw32-gcc
configure:4977: result: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile cl -nologo
configure:5246: checking for C compiler version
configure:5255: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile cl -nologo --version >&5
/c/users/eric/.conan/data/libgettext/0.20.1/_/_/build/e0c22822cdf05b624135e1696ae5cb784a23aeb3/source_subfolder/gettext-tools/configure: line 5257: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile: No such file or directory
configure:5266: $? = 127
configure:5255: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile cl -nologo -v >&5
/c/users/eric/.conan/data/libgettext/0.20.1/_/_/build/e0c22822cdf05b624135e1696ae5cb784a23aeb3/source_subfolder/gettext-tools/configure: line 5257: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile: No such file or directory
configure:5266: $? = 127
configure:5255: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile cl -nologo -V >&5
/c/users/eric/.conan/data/libgettext/0.20.1/_/_/build/e0c22822cdf05b624135e1696ae5cb784a23aeb3/source_subfolder/gettext-tools/configure: line 5257: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile: No such file or directory
configure:5266: $? = 127
configure:5255: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile cl -nologo -qversion >&5
/c/users/eric/.conan/data/libgettext/0.20.1/_/_/build/e0c22822cdf05b624135e1696ae5cb784a23aeb3/source_subfolder/gettext-tools/configure: line 5257: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile: No such file or directory
configure:5266: $? = 127
configure:5286: checking whether the C compiler works
configure:5308: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile cl -nologo -Zi -Ob0 -Od -MDd -FS -I/c/users/eric/.conan/data/libiconv/1.16/_/_/package/d057732059ea44a47760900cb5e4855d2bea8714/include -LIBPATH:/c/users/eric/.conan/data/libiconv/1.16/_/_/package/d057732059ea44a47760900cb5e4855d2bea8714/lib conftest.c iconv.lib charset.lib >&5
/c/users/eric/.conan/data/libgettext/0.20.1/_/_/build/e0c22822cdf05b624135e1696ae5cb784a23aeb3/source_subfolder/gettext-tools/configure: line 5310: /c/users/eric/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile: No such file or directory
configure:5312: $? = 127
```
it picked up the wrong path to `compile`, which is not in `C:\Users\eric\.conan\data\automake\1.16.3\_\_\package\3e48e69237f7f2196164383ef9dedf0f93cbf249\bin\share\automake-1.16` but in `C:\Users\eric\.conan\data\automake\1.16.3\_\_\package\3e48e69237f7f2196164383ef9dedf0f93cbf249\res\automake-1.16`

File: recipes/libgettext/all/conanfile.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def build(self):
             elif self.settings.arch == "x86_64":
                 host = "x86_64-w64-mingw32"
                 rc = "windres --target=pe-x86-64"
-            automake_perldir = tools.unix_path(os.path.join(self.deps_cpp_info['automake'].rootpath, "bin", "share", "automake-1.16"))
+            automake_perldir = tools.unix_path(os.path.join(self.deps_cpp_info['automake'].rootpath, "res", "automake-1.16"))
```
Comment (madebr) [inline recipes/libgettext/all/conanfile.py:95]: I think you can use the `AUTOMAKE_PERLLIBDIR` environment variable.
See https://github.com/conan-io/conan-center-index/blob/c25a202acf6bf13c9ca33b324f8e1f15bb21ea0d/recipes/automake/all/conanfile.py#L127-L129

File: recipes/libgettext/all/conanfile.py [hunk 2]
```diff
@@ -50,10 +50,10 @@ def requirements(self):
         self.requires("libiconv/1.16")
 
     def build_requirements(self):
-        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+        if self.settings.os == 'Windows' and not tools.get_env("CONAN_BASH_PATH"):
```
Comment (madebr) [inline recipes/libgettext/all/conanfile.py:53]: This needs to use `self.settings_build` (or a wrapper `self._settings_build`).

File: recipes/libgettext/all/conanfile.py [hunk 3]
```diff
@@ -50,10 +50,10 @@ def requirements(self):
...
             self.build_requires("msys2/cci.latest")
         if self._is_msvc:
-            self.build_requires("automake/1.16.2")
+            self.build_requires("automake/1.16.3")
```
Comment (madebr) [inline recipes/libgettext/all/conanfile.py:56]:
```suggestion
            self.build_requires("automake/1.16.4")
```

File: recipes/libgettext/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/libgettext/all/conanfile.py:95]: Or even better, use `self.deps_user_info_build.compile`:
https://github.com/conan-io/conan-center-index/blob/c25a202acf6bf13c9ca33b324f8e1f15bb21ea0d/recipes/automake/all/conanfile.py#L137

**Review by SSE4 — APPROVED** [review]

**Review by madebr — APPROVED** [review]

File: recipes/libgettext/all/conanfile.py [hunk 4]
```diff
@@ -49,11 +49,20 @@ def configure(self):
     def requirements(self):
         self.requires("libiconv/1.16")
 
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
```
Comment (ericriff) [inline recipes/libgettext/all/conanfile.py:55]: I won't make you lose the approvals, but this extra newlines trigger me lol

**Review by ericriff — APPROVED** [review]

**Pull request merged**